### PR TITLE
Add a test to check the actual behavior when cloning mocks (See #229)

### PR DESCRIPTION
**Pull request by jubianchi**
This is more a POC than an actual pull-request (even if I think that functional tests could be good to check bugfixes).

I tried to reproduce a use case @stephpy wrote in the #229 with a test. This test is green and if anyone can confirm that the bug is fixed, we can be all happy. 

This issue was here for a long time and @mageekguy spend a long time trying to figure out how to fix it. TOday it seems everything works as expected \o/

@atoum/all let me know if you have any other use case I should add to this test. 
Also let me know if you find those functional tests useful :)

### CODE REVIEW

File: classes/scripts/runner.php [hunk 1]
```diff
@@ -230,7 +230,11 @@ public function useConfigFile($path)
 
 	public function testIt()
 	{
-		$this->runner->addTestsFromDirectory(atoum\directory . '/tests/units/classes');
+		$this->runner
+            ->addTestsFromDirectory(atoum\directory . '/tests/units/classes')
```
Comment (Hywan) [inline classes/scripts/runner.php]: CS. Yes, tabs…

**Comment by Hywan** [comment]
Seems correct for me. @stephpy could confirm.

**Comment by jubianchi** [comment]
@mageekguy I just saw [your comment](https://github.com/atoum/atoum/issues/229#issuecomment-119852558) and I added a testcase here : `==` seems ok.

**Comment by mageekguy** [comment]
@jubianchi: VERY strange.

**Pull request merged**